src/chunk_append/exec.c

```diff
@@ -333,18 +333,44 @@ chunk_append_exec(CustomScanState *node)
 		 * more subplans, return the empty slot set up for us by
 		 * chunk_append_begin.
 		 */
-		if (state->runtime_exclusion)
-			state->current = bms_next_member(state->valid_subplans, state->current);
-		else
-			state->current++;
-
-		if (state->current >= state->num_subplans || state->current < 0)
-			return ExecClearTuple(node->ss.ps.ps_ResultTupleSlot);
+		choose_next_subplan(state);
 
 		/* Else loop back and try to get a tuple from the new subplan */
 	}
 }
 
+static void
+choose_next_subplan(ChunkAppendState *state)
+{
+	if (state->num_subplans == 0)
+	{
+		state->current = NO_MATCHING_SUBPLANS;
+		return;
+	}
+
+	if (state->runtime_exclusion)
+	{
+		if (!state->runtime_initialized)
+			initialize_runtime_exclusion(state);
+
+		if (bms_is_empty(state->valid_subplans))
```

Comment:
how come we do this now even if `state->runtime_initialized`?

Reply:
runtime exclusion might have excluded all but bms_next_member should come to the same conclusion so this check can be removed